lib/main.dart

```diff
@@ -56,12 +56,24 @@ class _AppState extends State<App> {
     //TODO: wait until settings is ready?
     settings.onChange().listen((_) {
       setState(() {
-        if (!settings.useSystemColors) {
+        if (settings.useSystemColors) {
+          brightness = SchedulerBinding.instance.platformDispatcher.platformBrightness;
+        } else {
           brightness = settings.darkMode ? Brightness.dark : Brightness.light;
         }
       });
     });
 
+    // Listen to changes to the system brightness mode, update accordingly
+    var dispatcher = SchedulerBinding.instance.platformDispatcher;
```

Comment:
Can this be `val`? I know you are modifying the instance member, but the memory address of the object stored at `dispatcher` doesn't change. Genuine question, I'm not sure how Dart behaves here.

Reply:
Uh, I must be thinking of Kotlin. Dart doesn't seem to have val?

Reply:
Oh it'd be `final`. But meh.